core/tests/test_utils.py

```diff
@@ -2267,6 +2267,26 @@ def post_json(
 
         return self._parse_json_response(json_response, expect_errors)
 
+    def post_json_with_custom_body(
```

Comment:
Why can't this be combined with post_json? I don't know what a "custom body" is ... it might require more explanation.